tools/regalloc/regalloc.ml

```diff
@@ -0,0 +1,82 @@
+(* This program simply runs a register allocator on the all the functions saved
+   in a .cmir-cfg-regalloc file. *)
+
+module List = ListLabels
+
+type register_allocator =
+  | GI
+  | IRC
+  | LS
+
+let allocators = ["gi", GI; "irc", IRC; "ls", LS]
+
+external time_include_children : bool -> float
+  = "caml_sys_time_include_children"
+
+let cpu_time () = time_include_children false
+
+let process_function (cfg_with_layout : Cfg_with_layout.t)
+    (register_allocator : register_allocator) =
+  Printf.eprintf "  processing function %S...\n%!"
+    (Cfg_with_layout.cfg cfg_with_layout).fun_name;
+  let add_regs (acc : Reg.Set.t) (instr : _ Cfg.instruction) : Reg.Set.t =
+    let acc = Reg.add_set_array acc instr.arg in
+    let acc = Reg.add_set_array acc instr.res in
+    acc
+  in
+  let all_regs =
+    Cfg_with_layout.fold_instructions cfg_with_layout ~init:Reg.Set.empty
+      ~instruction:add_regs ~terminator:add_regs
+  in
+  let relocatable_regs = Reg.Set.filter Reg.is_unknown all_regs in
+  Printf.eprintf "    %d register(s)...\n%!" (Reg.Set.cardinal relocatable_regs);
+  Reg.For_testing.set_state
+    ~stamp:(succ (Reg.Set.max_elt relocatable_regs).stamp)
+    ~relocatable_regs:(List.of_seq (Reg.Set.to_seq relocatable_regs));
```

Comment:
Is this part intended to guarantee that the "shared state" is the same as in the original compilation? Instead of trying to reverse-engineer this information, can we just save it alongside the cfg everything we need it? Do we need to save the label too somewhere?  It's saved as part of Cfg file format, for the purpose of "start from emit" it only need to not overlap with existing labels, but maybe we want something more precise here for regalloc?